src/LIBSVM.jl

```diff
@@ -185,7 +185,7 @@ function instances2nodes(instances::AbstractMatrix{<:Real})
     for i=1:ninstances
         k = 1
         for j=1:nfeatures
-            nodes[k, i] = SVMNode(Int32(j), Float64(instances[j, i]))
+            nodes[k, i] = SVMNode(Int32(j-1), Float64(instances[j, i]))
```

Comment:
I also think LIBSVM use one-base indexing, according to its manual.
So what is the point of settiing `SVMNode.index` with `j-1`?

Reply:
The lines of the input file that `svm-train` accepts should be in the form `<label> 0:i 1:K(xi,x1) ... L:K(xi,xL)`, such as
```
15  0:1 1:4 2:6  3:1
45  0:2 1:6 2:18 3:0
25  0:3 1:1 2:0  3:1
```
However, it seems to be working correctly without the `-1` too. So stick to the original version?

Reply:
hmm, seems it requires index start from 1 for normal case, precomputed kernel is the only special case with index starting from 0.

Ref: https://github.com/cjlin1/libsvm/blob/eedbb147ea79af44f2ecdca1064f2c6a8fe6462d/README#L76-L80

Reply:
I think it is only a convention for the input files. 

LIBSVM's `svm-train` requires the zero-indexed column to contain instance indices, otherwise it throws an error. However, once the input file is successfully parsed and `svm_node`s are produced, LIBSVM does not use the indices except for model (de)serialization.

Reply:
> I think it is only a convention for the input files.

oh, okay.
I don't read the LIBSVM code yet. 
If you already confirmed that by reading the LIBSVM internal code base, how about putting some comments in our code block to point out the difference between LIBSVM's doc and our implementation?
Maybe that will help the other contributors.